drivers/at86rf2xx/at86rf2xx_getset.c

```diff
@@ -475,6 +475,14 @@ uint8_t at86rf2xx_set_state(at86rf2xx_t *dev, uint8_t state)
              old_state == AT86RF2XX_STATE_IN_PROGRESS);
 
     if (state == AT86RF2XX_STATE_FORCE_TRX_OFF) {
+        if (old_state == AT86RF2XX_STATE_TX_ARET_ON) {
+            /* When on TX_ARET_ON, go to idle state before going to TRX_OFF */
+            assert(dev->pending_tx != 0);
+            if ((--dev->pending_tx) == 0) {
+                at86rf2xx_set_state(dev, dev->idle_state);
```

Comment:
My interpretation of the data sheet is that the FORCE_TRX_OFF state should be used to abort any ongoing transfers in the middle of everything and move the transceiver to TRX_OFF to ensure that the state machine is in a known state before changing settings or modes. I think this change will add unexpected waits for TX completion in places where no wait is expected. Perhaps it is the usage of FORCE_TRX_OFF which is wrong? What were the problems that you were seeing in your application? Did the pending_tx counter get stuck at 1 and never dropping back to 0?
Perhaps we could just reset that counter instead of waiting for TX completion?

Reply:
I did this because transition from AT86RF2XX_STATE_TX_ARET_ON to AT86RF2XX_STATE_SLEEP failed. It's not a common case because most of the time, transceiver is in IDLE state before going to SLEEP state.
When I say it fails, it means that transceiver does not go to sleep and then when we wake it up, it doesn't.

Each time transceiver succeed to send a frame, it raises an IRQ and the driver changes the state of the transceiver to idle_state before continue the execution.
So i did the same outside the IRQ which works for me 
You can see what driver does in:
static void _isr(netdev_t *netdev)

I didn't check if pending_tx counter gets stuck

Reply:
What do you suggest?